src/core/Buttons/DropdownButton/DropdownButton.tsx

```diff
@@ -43,14 +44,16 @@ export const DropdownButton: React.FC<DropdownButtonProps> = (props) => {
 
   const [isMenuOpen, setIsMenuOpen] = React.useState(false);
 
+  const windowSize = useWindowSize(500);
+
   const [menuWidth, setMenuWidth] = React.useState(0);
   const ref = React.useRef<HTMLButtonElement>(null);
 
   React.useEffect(() => {
     if (ref.current) {
       setMenuWidth(ref.current.offsetWidth);
     }
-  }, [children, size, styleType]);
+  }, [children, size, styleType, windowSize]);
```

Comment:
Is this really needed? Cause I looked at other libs and no one is doing this. Also, it just adds additional complexity.

Reply:
Hmm idk, it's nice. Which libs did you check? Most of them are meant to be as customizable as possible, rather than enforcing standards.

I guess we could also just provide a util.

https://material-ui.com/components/use-media-query/
https://material-ui.com/components/textarea-autosize/
https://blueprintjs.com/docs/#popover2-package/resize-sensor2
https://material.angular.io/cdk/layout/overview

Reply:
antd and material-ui changes the select element and menu stays the same size when resizing. react-bootstrap and blueprint does not change anything on resize so hard to tell.

Reply:
Material likes to keep the dropdown width same as select, but it does not have any example where select is taking full width.
![image](https://user-images.githubusercontent.com/9084735/119504965-7abea780-bd3a-11eb-9a11-f91c5ba56ce4.png)

I would think if you were to code a dynamic (resizable) dropdown using Material, you'd want the width to respond to the change.

But AntD does have some examples.
![antd-select](https://user-images.githubusercontent.com/9084735/119502252-be63e200-bd37-11eb-958d-f874623efcd6.gif)

Reply:
Oh this is interesting: [https://popper.js.org/docs/v2/modifiers/community-modifiers/#:~:text=samewidth%20modifier%3A%20make%20your%20popper%20the%20same%20width%20as%20the%20reference.](https://popper.js.org/docs/v2/modifiers/community-modifiers/)

Reply:
Your AntD gif does the same as we do right now 😀 and if understand correctly your hook will work with opened menu and resized window.
That modifier would solve a lot of duplication. But does it handle live resizes?

Reply:
This is our current behavior and really the main thing I'm trying to solve.
![resize](https://user-images.githubusercontent.com/9084735/119663639-84124780-be00-11eb-9617-775ea19827a8.gif)

And I haven't gotten that modifier to work tbh.

Reply:
I don't think this is a bad behavior:
![image](https://user-images.githubusercontent.com/36186912/119800224-7a164480-bee5-11eb-9d3d-9024815e40eb.png)
Maybe @FlyersPh9 can comment on this one.